Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM maven:3.6.3-jdk-8 as builder
+WORKDIR /build
+# Coping the pom file
+COPY pom.xml .
+# Download depedencies for cache
+RUN mvn org.apache.maven.plugins:maven-dependency-plugin:3.1.1:go-offline -B
+# Go offline command doesn't download this artifact, so i am downloading manually
+RUN mvn org.apache.maven.plugins:maven-dependency-plugin:3.1.1:get \
+        -DrepoUrl=https://jcenter.bintray.com/ \
+        -Dartifact=org.apache.maven.surefire:surefire-junit4:2.12.4
+# Coping source files
+COPY src/ /build/src
+# Building the project
+RUN mvn -o -B package
```

Comment:
You can configure to only automatic build a specific tag regex in docker hub, or you can also build manually and push the image yourself.

Reply:
Building from source isn't a supported method for obtaining a jar for this bot (I don't provide build scripts), so for a docker release that's provided in this repo, I'd only want it to check for and download the latest jar.

Reply:
I think is more reliable to build from source, is there anything special about the building process? Basically i just need to run the mvn package command, is that right?

Reply:
Building from source is less reliable because there is no guarantee that what is on the repo at any given point works or builds, and there could be work-in-progress features or changes. The released jars are the only thing guaranteed to run correctly.

Reply:
If we just build release tags in has the same reliability.

Reply:
Release tags don't necessarily precisely match the released jars, I occasionally build before pushing the latest changes, or vice-versa, so building from source is not reliable (and not supported). The different platform builds are manually packaged as well.

Reply:
I don't understand why building from source is not supported in this project, especially a maven project, where you can build the jar with a simple command. 
I prefer to build from source instead of blindly trust in provided binaries, and i think most of the people agree with me.

Reply:
You're free to build from source if you desire. However, only the released builds are what I provide support for. It is very time-consuming and frustrating to try to debug/deal with problems from builds other than the ones I provide. [Additionally, this repository is open-source to allow contributions and for other developers to see how the bot operates, not as a build source.](https://github.com/jagrosh/MusicBot#editing)

I do not want someone that follows the docker instructions provided in the wiki to end up with a non-supported version of the bot. Therefore, any docker functionality added to this repository should download a supported build, not build from source.

Reply:
If you would just built the exact commit you tagged and do reproducible builds we wouldn't have that discussion and this fact makes it impossible to ever land in any distro as they would always built that tagged commit.